refinery/data_set_manager/tasks.py

```diff
@@ -332,13 +341,25 @@ def generate_bam_index(auxiliary_file_store_item_uuid, datafile_path):
     # fail if we can't get what we want.
     bam_index_file_extension = FileExtension.objects.get(name="bai").name
     auxiliary_file_store_item = FileStoreItem.objects.get(
-        uuid=auxiliary_file_store_item_uuid)
+        uuid=auxiliary_file_store_item_uuid
+    )
 
     # Leverage pysam library to generate bam index file
     # FIXME: This should be refactored once we don't have a need for
     # Standalone IGV because this is creating a bam_index file in the same
     # directory as it's bam file
-    pysam.index(bytes(datafile_path))
+    if settings.REFINERY_S3_USER_DATA:
+        key = datafile_path
+        bucket = settings.MEDIA_BUCKET
+        temp_file = os.path.join(tempfile.gettempdir(), key)
+        os.makedirs(os.path.abspath(os.path.join(temp_file, os.pardir)))
+        with open(temp_file, 'wb') as destination:
+            download_s3_object(bucket, key, destination)
+        pysam.index(bytes(temp_file))
+        datafile_path = temp_file
+        os.remove(temp_file)
+    else:
+        pysam.index(bytes(datafile_path))
```

Comment:
The index file should not be written directly to the file store dir - it is part of the reason this code is broken (see also line 348).

Reply:
maybe i am confused by what you mean by "file store dir" - this file is downloaded into a temporary directory created by `tempfile.gettempdir()`, which, as far as i can tell, is not the "file store dir." pysam then dumps the file into there.  `pysam.index` can take a string argument to tell it where to dump, but i'm not sure what is better than a temporary directory.: 

https://pysam.readthedocs.io/en/latest/usage.html#using-samtools-commands-within-python

http://www.htslib.org/doc/samtools.html

Reply:
i don't see why not - will add it in